enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,428 @@
+or out of bounds can cause problems such as extremely poor performance or start
+up errors. Mitigation of this scenario will be to provide proper guidance and
+guidelines for customers who enable this enhancement. Furthermore, the
+performance team would need to be reached out to for more specific information
+around upper and lower bounds of CPU sets for running an Openshift cluster.
+
+It is possible to build a cluster with the feature enabled and then add a node
+in a way that does not configure the workload partitions only for that node. We
+do not support this configuration as all nodes must have the feature turned on.
+However, there might be a race condition where a node is added and is in the
+process of being restarted with workload partitioning, during that process pods
+being admitted will trigger a warning. We expect the resulting error message
+described in [failure modes](#failure-modes) to explain the problem well enough
+for admins to recover.
+
+A possible risk are cluster upgrades, this is the first time this enhancement
+will be for multi-node clusters, we need to run more tests on upgrade cycles to
+make sure things run as expected.
+
+### Drawbacks
+
+This feature contains the same drawbacks as the [Management Workload
+Partitioning](management-workload-partitioning.md).
+
+Several of the changes described above are patches that we may end up carrying
+downstream indefinitely. Some version of a more general "CPU pool" feature may
+be acceptable upstream, and we could reimplement management workload
+partitioning to use that new implementation.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+We will add a CI job with a cluster configuration that reflects the minimum of
+2CPU/4vCPU masters and 1CPU/2vCPU worker configuration. This job should ensure
+that cluster deployments configured with management workload partitioning pass
+the compliance tests.
+
+We will add a CI job to ensure that all release payload workloads have the
+`target.workload.openshift.io/management` annotation and their namespaces have
+the `workload.openshift.io/allowed` annotation.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in
+  [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
```

Comment:
We need to cover here how we will deal with deployed SNO's pre-4.12. Reinstalling will not be an option

Reply:
Would something need to recognize that partitioning is enabled and create the performance profile resource(s) correctly? If so, we might need to add that behavior in 4.11 to ensure a smooth upgrade.

Reply:
In regard to this, the NTO can be used to identify that it's been set via the `capacity` field we apply to the nodes, we can use that information to set the global flag and make sure we don't apply a new machine config unless the user explicitly sets the new `PerformanceProfile` enum, that way we should avoid any unwanted restarts. 

This way we can offer two options to those customers, if they wish to use the new variable and not manage the separate machine config, they can set the Enum and incur a restart so the NTO takes care of things, or leave it alone and not worry about it. However, If they change the partition size later on, we can take that opportunity to consolidate the machine configs and clean up. 

I feel like that's a reasonable approach for SNO pre 4.12. What do you guys think?

Reply:
Would we only support both configuration modes on existing SNO deployments? Or would users be able to deploy new SNO instances using the old configuration approach so that we have to carry that upgrade logic indefinitely?